### PR TITLE
Add support for akka streams in spring mvc #516

### LINKED ISSUE #516: Provide support for Akka Streams in Spring MVC, because why not
Since they also implement reactive streams it's easy to do :) 
*RS in Action!*

### PR DESCRIPTION
**Pull request by ktoso**
Resolves #516

Made sure the naming makes sense.
Please sanity check things.

I did not figure out the "magically add config", which is the auto configuration things.
But if people want that they can help out I guess :)

### CODE REVIEW

File: docs/src/main/paradox/spring-web.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Spring Web
+
+Spring 5.0 introduced compatibility with the [Reactive Streams](https://github.com/reactive-streams/reactive-streams-jvm)
```
Comment (raboof) [inline docs/src/main/paradox/spring-web.md]: Maybe link to http://www.reactive-streams.org/ for this audience?

File: docs/src/main/paradox/spring-web.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Spring Web
+
+Spring 5.0 introduced compatibility with the [Reactive Streams](https://github.com/reactive-streams/reactive-streams-jvm) 
+interfaces, that Lightbend co-lead with *Akka Streams* along with other library, such as Pivotal.
```
Comment (raboof) [inline docs/src/main/paradox/spring-web.md]: perhaps ", a library interoperability standardization effort co-lead by Lightbend (with Akka Streams) along with Kaazing, Netflix, Pivotal, Red Hat, Twitter and many others" ?

File: docs/src/main/paradox/spring-web.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+Which will make Spring aware of Akka Streams types and their relation to their respective Reactive Streams types. From then on you can, for example, simply return a `Source `(regardless if it's `javadsl` or `scaladsl`) in an Spring mvc endpoint:
+ 
+Java
+: @@snip (../../../../spring-web/src/test/java/akka/stream/alpakka/spring/web/SampleController.java) { #use }
+
+
+
+## Shameless plug: Akka HTTP 
+
+While the integration presented here works, it's not quite the optimal way of using Akka n conjunction with serving HTTP apps.
```
Comment (raboof) [inline docs/src/main/paradox/spring-web.md]: missing 'i'

File: docs/src/main/paradox/spring-web.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+If you're new to reactive systems and picking technologies, you may want to have a look at [Akka HTTP](https://doc.akka.io/docs/akka-http/current/scala/http/).
+
+If you decided, or are forced to, use Spring MVC this integration should help you achieve the basic streaming scenarios though.
```
Comment (raboof) [inline docs/src/main/paradox/spring-web.md]: Hmm maybe remove 'or are forced to'

Comment (ktoso) [reply docs/src/main/paradox/spring-web.md]: hehe okey :)

File: docs/src/main/paradox/spring-web.md (hunk 1 above)
Comment (ktoso) [reply docs/src/main/paradox/spring-web.md]: yeah, good point

File: docs/src/main/paradox/spring-web.md (hunk 2 above)
Comment (ktoso) [reply docs/src/main/paradox/spring-web.md]: Yeah very good point, to re-stress the many-people involved and de-emphasise pivotal

**Comment by ktoso** [comment]
Addressed comments

**Comment by ktoso** [comment]
I'll merge, happy to address any further comments though.
I'll also make a ticket so maybe someone wants to pick up the "auto config" part of it https://docs.spring.io/spring-boot/docs/current/reference/html/using-boot-auto-configuration.html

**Pull request merged**